cloudpathlib/cloudpath.py

```diff
@@ -866,3 +892,81 @@ def _resolve(path: PurePosixPath) -> str:
         newpath = newpath + sep + name
 
     return newpath or sep
+
+
+# These objects are used to wrap CloudPaths in a context where we can use
+# the python pathlib implementations for `glob` and `rglob`, which depend
+# on the Selector created by the `_make_selector` method being passed
+# an object like the below when `select_from` is called. We implement these methods
+# in a simple wrapper to use the same glob recursion and pattern logic without
+# rolling our own.
+#
+# Designed to be compatible when used by these selector implementations from pathlib:
+# https://github.com/python/cpython/blob/3.10/Lib/pathlib.py#L385-L500
+class _CloudPathSelectableAccessor:
+    def __init__(self, scandir_func):
+        self.scandir = scandir_func
+
+
+class _CloudPathSelectable:
+    def __init__(
+        self, relative_cloud_path: PurePosixPath, children: List, is_dir: bool, exists: bool = True
+    ):
+        self._path = relative_cloud_path
+        self._all_children = children
+
+        self._accessor = _CloudPathSelectableAccessor(self.scandir)
+
+        self._is_dir = is_dir
+        self._exists = exists
+
+    def __repr__(self):
+        return str(self._path)
+
+    def is_dir(self):
+        return self._is_dir
+
+    def exists(self):
+        return self._exists
+
+    def is_symlink(self):
+        return False
+
+    @property
+    def name(self):
+        return self._path.name
+
+    @staticmethod
+    @contextmanager
+    def scandir(root):
+        results = []
+        for c, is_dir in root._all_children.items():
+            if c.parent == root._path:
+                results.append(root._make_child_relpath(c.name))
+        yield results
```

Comment:
```suggestion
        yield (
            root._make_child_relpath(c.name)
            for c, _ in root._all_children.items()
            if c.parent == root._path
        )
```

I think this should also work? Not sure if having a generator expression is clearer than the for loop, or if turning the list into an iterator is a meaningful optimization.